Testscripts/Linux/validate-intel-sgx-driver.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+#######################################################################
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+#
+# validate-intel-sgx-driver.sh
+# Description:
+#    Check the existence of Intel sgx driver.
+#    Install Open-Enclave and build&run Samples
+# Supported Distros:
+#    Ubuntu
+#######################################################################
+
+set -e
+set -x
+
+. utils.sh || {
+    echo "Error: unable to source utils.sh!"
+    echo "TestAborted" >state.txt
+    exit 1
+}
+
+UtilsInit
+
+sudo apt-get update
+sudo apt-get -y upgrade
+echo 'deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu bionic main' | sudo tee /etc/apt/sources.list.d/intel-sgx.list
+wget -qO - https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | sudo apt-key add -
+
+echo "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-7 main" | sudo tee /etc/apt/sources.list.d/llvm-toolchain-bionic-7.list
+wget -qO - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+
+echo "deb [arch=amd64] https://packages.microsoft.com/ubuntu/18.04/prod bionic main" | sudo tee /etc/apt/sources.list.d/msprod.list
+wget -qO - https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -
+sudo apt-get update
+
+LogMsg "----- Checking sgx driver -----"
+if ! modinfo intel_sgx; then
```

Comment:
Adding [] is wrong.

Reply:
Francis, @lubaihua33 is talking about programming style in common practice in the LISAv2.  As you see other bash script, we try to make consistent manner of writing. Can you add [ ] in if-statement? If this is "wrong", please put your response here.

Reply:
"**[ condition-to-test-for ]**" is the syntax for **conditional expression** [man page](http://manpages.ubuntu.com/manpages/eoan/en/man1/test.1.html). There is no direct way of using `modinfo intel_sgx` as an expression. Here, I want to test if the command `modinfo intel_sgx` succeeded or failed, which checks the exit code of the command. `if COMMAND` is the syntactically cleanest way to perform such condition check ([further reading fyi](http://mywiki.wooledge.org/BashGuide/TestsAndConditionals#Conditional_Blocks_.28if.2C_test_and_.5B.5B.29)). If you search LISAv2 code base, this type of coding style is used in almost all similar situations. If you insist using the [] syntax, the code should be rewritten to:

```
modinfo intel_sgx
if [ $? -ne 0 ]; then
    echo ...
```

which is verbose at least. However, there is another hidden issue besides verbose. Since `set -x` is specified in this script, the script will exit when any line of expression fails. So, if `modinfo intel_sgx` fails, the script will quit right after, which isn't what I want -- total control over the code flow.